src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -83,34 +94,65 @@ public ServiceBusJmsConnectionFactory(ConnectionStringBuilder connectionStringBu
      */
     public ServiceBusJmsConnectionFactory(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
         this.settings = settings;
-        this.initialize(sasKeyName, sasKey, host, settings);
+        this.password = sasKey;
+        this.userName = sasKeyName;
+        this.host = host;
+        this.initializeConnectionString();
     }
     
-    private void initialize(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
-        if (sasKeyName == null || sasKeyName == null || host == null) {
-            throw new IllegalArgumentException("SAS Key, SAS KeyName and the host cannot be null for a ServiceBus connection factory.");
-        }
-        
-        if (settings == null) {
-            settings = new ServiceBusJmsConnectionFactorySettings();
-        }
-        
-        if (this.builder == null) {
+    /**
+     * Create a ServiceBusJmsConnectionFactory using a credential and host name.
+     * @param Credential. A token provider credential that will be used to acquire an aad token.
+     * @param host. The host name of the ServiceBus namespace. Example: your-namespace-name.servicebus.windows.net
+     * @param settings The options used for this ConnectionFactory. Null can be used as default.
+     * @throws Exception 
+     */
+    public ServiceBusJmsConnectionFactory(TokenCredential credential, String host, ServiceBusJmsConnectionFactorySettings settings){
+    	this.settings = settings;
+    	this.aadAuthentication = new AadAuthentication(credential);
+    	this.userName = AAD_TOKEN_USERNAME;
+    	this.password = this.aadAuthentication.getAadToken();
+    	this.host = host;	
+        this.initializeAdd();
+    }
+    
+    private void initializeAdd() {
+    	this.initialize(this.userName, this.password, this.host, this.settings);
+    	this.setExtensionsForAad();
+    	this.initialized = true;
+    }
+    
+    private void initializeConnectionString() {
+
+    	this.initialize(this.userName, this.password, this.host, this.settings);
```

Comment:
nit: can this line move into the common initialize() method()? since its done for both similarly?

Reply:
this is calling the initialize method